routes/routes-api.js

```diff
@@ -0,0 +1,203 @@
+  db.Article.find({}).sort({dateTime: -1}).then(function(resp, err) {
+    if(err) {
+      console.log(err);
+    }
+    else {
+      res.json(resp);
+    }
+  })
+});
+
+
+// Route for grabbing a specific Article by id, populate it with it's note
+app.get("/articles/:id", function(req, res) {
+  db.Article.findOne({'_id' : req.params.id})
+  .populate("note")
+  .then(function(article) {
+    res.json(article);
+  })
+  .catch(function(err) {
+    res.json(err)
+  })
+});
+
+// Route for saving/updating an Article's associated Note
+app.put("/articles/:id", function(req, res) {
+  db.Article.update({externalArticleId : req.params.id}, {isSaved : true})
+  .then(function(dbArticle) {
+    res.json(dbArticle);
+  }).catch(function(err) {
+    res.json(err);
+  })
+});
+
+// Route for saving/updating an Article's associated Note
+app.put("/articles-del/:id", function(req, res) {
+  db.Article.update({externalArticleId : req.params.id}, {isSaved : false})
+  .then(function(dbArticle) {
+    res.json(dbArticle);
+  }).catch(function(err) {
+    res.json(err);
+  })
+});
+
+// Route for saving a new Note to the db and associating it with a User
+app.put("/articles-add-note/:id", function(req, res) {
+
+  // Create a new Note in the db
+  db.Note
+    .create({body: req.body.body})
+    .then(function(dbNote) {
+      // If a Note was created successfully, find one User (there's only one) and push the new Note's _id to the User's `notes` array
+      // { new: true } tells the query that we want it to return the updated User -- it returns the original by default
+      // Since our mongoose query returns a promise, we can chain another `.then` which receives the result of the query
+      return db.Article.findOneAndUpdate({externalArticleId : req.params.id}, { $push: { notes: dbNote._id } }, { new: true });
+    })
+    .then(function(dbArticle) {
+      // If the User was updated successfully, send it back to the client
+      console.log(dbArticle)
+      res.json(dbArticle);
+    })
+    .catch(function(err) {
+      // If an error occurs, send it back to the client
+      console.log(err)
+      res.json(err);
+    });
+});
+
+// Route to get all User's and populate them with their notes
+app.get("/populated", function(req, res) {
+  // Find all users
+  db.User
+    .find({})
+    // Specify that we want to populate the retrieved users with any associated notes
+    .populate("notes")
+    .then(function(dbUser) {
+      // If able to successfully find and associate all Users and Notes, send them back to the client
+      res.json(dbUser);
+    })
+    .catch(function(err) {
+      // If an error occurs, send it back to the client
+      res.json(err);
+    });
+});
+
+app.delete("/note-del/:id", function(req, res) {
+  // Find all users
+  db.Note.findOneAndRemove({
+      "_id": req.params.id
+    }, function(err) {
+      if (err) {
+        console.log(err);
+        res.send(err);
+      } else {
```

Comment:
this doesn't need to be wrapped in an if, it only succeeds in costricting your coding space more.